src/components/FileBrowser.vue

```diff
@@ -0,0 +1,285 @@
+          const resp = await gr.get(endpoint, { params });
+          if (resp.status !== 200) return [];
+          return resp.data;
+        };
+        this.rowsLoading = true;
+        const rows = [
+          ...(await request(this.girderRest, GIRDER_FOLDER_ENDPOINT, this.folderParams)),
+          ...(await request(this.girderRest, GIRDER_ITEM_ENDOINT, this.itemParams)),
+        ].map(item => ({
+          name: item.name,
+          type: item._modelType,
+          id: item._id,
+          size: item.size ? this.formatSize(item.size) : '',
+          icon: item._modelType in ICON_MAP ? ICON_MAP[item._modelType] : 'file',
+        }));
+        this.rowsLoading = false;
+        return rows;
+      },
+      watch() {
+        return [
+          this.folderParams,
+          this.itemParams,
+          this.counts,
+          this.location,
+          this.refreshCounter_,
+        ];
+      },
+    },
+  },
+  watch: {
+    location(newval, oldval) {
+      // force reset pagination when location changes.
+      if (newval.id !== oldval.id) {
+        this.pagination.page = 1;
+      }
+    },
+    selected(newval) {
+      this.$emit('selection-changed', newval);
+    },
+  },
+  computed: {
+    folderParams() {
+      return {
+        parentType: this.location.type,
+        parentId: this.location.id,
+        limit: this.pagination.rowsPerPage,
+        offset: (this.pagination.page - 1) * this.pagination.rowsPerPage,
+      };
+    },
+    itemParams() {
+      const limit = this.counts.nFolders > this.folderParams.offset
+        ? this.pagination.rowsPerPage - (this.counts.nFolders - this.folderParams.offset)
+        : this.pagination.rowsPerPage;
+      const offset = this.folderParams.offset > 0
+        ? this.folderParams.offset - this.counts.nFolders
+        : 0;
+      if (limit <= 0 || (this.location.type !== 'folder')) {
+        return null; // No items should be shown on the current page.
+      }
+      return { folderId: this.location.id, limit, offset };
+    },
+    loading() { return this.rowsLoading || this.breadcrumbLoading; },
+    totalItems() { return this.counts.nFolders + this.counts.nItems; },
+  },
+  methods: {
+    toggleAll() {
+      if (this.selected.length === this.rows.length) {
+        this.selected = [];
+      } else {
+        this.selected = this.rows.slice();
+      }
+    },
+    changeLocation(item) {
+      const { type, id, name } = item;
+      if ((this.location.id !== id || this.location.type !== type)
+          && type !== 'item') {
+        this.$emit('update:location', { type, id, name });
+      }
+    },
+    refresh() {
+      this.refreshCounter_ += 1;
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+v-data-table.girder-file-browser-component.elevation-1(
+    select-all,
+    :headers-length="4",
+    v-model="selected",
+    :pagination.sync="pagination",
+    :items="rows",
+    :total-items="totalItems",
+    :loading="loading ? 'accent' : false",
```

Comment:
This one is totally on Vuetify. But maybe we could overwrite the style so that the loading indicator doesn't push the content down and up when being shown and invisible.

Reply:
I looked into that.  Because of where it's placed in the table, there's just no way to do this.  I'd have to write a new loader that puts the content of the loader into the table header then position it absolutely to keep it from taking up space on the page. 

I'm actually pretty miffed that Vuetify would do something this silly.

Reply:
It was not straight forward, but seems adding 
```
.girder-file-browser-component .v-datatable__progress .v-progress-linear
  position absolute
```
Would do the trick